agent/client/client.go

```diff
@@ -25,8 +25,8 @@ type Client struct {
 	sessionConfig *SessionConfig
 	done          chan bool
 
-	triggerListener func(*proto.TriggerRequest) error
-	pollListener    func(*proto.PollingRequest) error
+	triggerListener func(context.Context, *proto.TriggerRequest) error
```

Comment:
I'm adding these contexts so we can add instrumentation to the agent as well. It will be important to be able to collect metrics and some traces from agents if users allow it